src/ripple/core/sql_backend/SQLInterface_sqlite.cpp

```diff
@@ -0,0 +1,1280 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/ledger/AcceptedLedger.h>
+#include <ripple/app/ledger/LedgerMaster.h>
+#include <ripple/app/ledger/LedgerToJson.h>
+#include <ripple/app/ledger/TransactionMaster.h>
+#include <ripple/app/misc/Manifest.h>
+#include <ripple/basics/BasicConfig.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/core/DatabaseCon.h>
+#include <ripple/core/SQLInterface.h>
+#include <ripple/core/SQLInterface_shards.h>
+#include <ripple/core/SociDB.h>
+#include <ripple/json/to_string.h>
+#include <ripple/nodestore/DatabaseShard.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/range/adaptor/transformed.hpp>
+#include <soci/sqlite3/soci-sqlite3.h>
+
+namespace ripple {
+
+class SQLInterface_sqlite : public SQLInterface
+{
+public:
+    SQLInterface_sqlite(Application& app) : app_(app)
+    {
+    }
+
+    virtual void
+    addDBs(LedgerIndex shardIndex, DatabaseCon* tx, DatabaseCon* lgr) override;
+
+    virtual void
```

Comment:
We usually leave off `virtual` when we add `override`.

Reply:
Done.